### ISSUE TITLE
Divi/WooCommerce/Lifter Compatibility Issue

### ISSUE THREAD

**Issue by nrherron92** [gocodebox/lifterlms#1079]
### Reproduction Steps

+ Install and activate most recent version of Divi
+ Install and activate WooCommerce
+ Install and activate LifterLMS (without wc integration)
+ Make sure Divi builder is turned on for Courses post type
+ It's also a little easier to see this issue if you turn on Divi's setting for classic editor but it shows in Gutenberg too
+ Go into a course and choose edit with Divi. In the classic editor the builder will not load on the backend. In Gutenberg it stays in the code view and will not activate the Divi builder. The page will just continue loading on the front end builder.

### Expected Behavior

+ The page loads in the builder.

### Actual Behavior

+ With WooCommerce active, Divi seems to be calling the product filter as if our course is part of WooCommerce causing a conflict where the page won't load.

### Error Messages / Logs

+ Include any relevant error messages or log files
```
[03-Apr-2020 10:33:59 UTC] PHP Fatal error:  Uncaught Error: Call to undefined method LLMS_Product::has_attributes() in /Users/natalie/Web/fresh/wp-content/plugins/woocommerce/includes/wc-template-functions.php:1749
Stack trace:
#0 /Users/natalie/Web/fresh/wp-includes/class-wp-hook.php(287): woocommerce_default_product_tabs(Array)
#1 /Users/natalie/Web/fresh/wp-includes/plugin.php(206): WP_Hook->apply_filters(Array, Array)
#2 /Users/natalie/Web/fresh/wp-content/themes/Divi/includes/builder/functions.php(1178): apply_filters('woocommerce_pro...', Array)
#3 /Users/natalie/Web/fresh/wp-content/themes/Divi/includes/builder/functions.php(1319): et_fb_woocommerce_tabs()
#4 /Users/natalie/Web/fresh/wp-content/themes/Divi/includes/builder/frontend-builder/helpers.php(279): et_fb_current_page_params()
#5 /Users/natalie/Web/fresh/wp-content/themes/Divi/includes/builder/frontend-builder/helpers.php(2225): et_fb_get_dynamic_backend_helpers()
#6 /Users/natalie/Web/fresh/wp-includes/class-wp-hook.php(287): et_fb_backend_helpers('')
#7 /Users/natalie/Web/fresh/wp-incl in /Users/natalie/Web/fresh/wp-content/plugins/woocommerce/includes/wc-template-functions.php on line 1749

```

HS-115928
HS-116327
HS-117152

**Comment by thomasplevy** [comment]
Will this happen on any other post types or just courses?

**Comment by nrherron92** [comment]
It happens on courses and memberships. Divi loads on lessons and certificates with Woo on normally

**Comment by eri-trabiccolo** [comment]
@thomasplevy courses and memberships, our products.
I've found in the Divi code where this happens.

As already happened with another plugin, their code is just checking if the global `$product` is set, and assumes that's a wc product. Then it calls a WooCommerce filter, which, once again, assumes that the global $product contains a wc product.

**Comment by thomasplevy** [comment]
Legacy code is going to kill me ..

**Comment by eri-trabiccolo** [comment]
Contact et devs would be great, any chance?

**Comment by thomasplevy** [comment]
Working on it

**Comment by thomasplevy** [comment]
@eri-trabiccolo could you post relevant lines of code in Divi where this check is being made please

**Comment by eri-trabiccolo** [comment]
sure, well, better to link the function where a check is missing:
The function is: `et_fb_woocommerce_tabs()`, located in Divi/includes/builder/functions.php

To me (tested) this line (line 1165 of that file):
`	if ( ! isset( $product ) && et_is_woocommerce_plugin_active() ) ) {
`
can be turned into
`	if ( ! isset( $product ) && et_is_woocommerce_plugin_active() || ! wc_get_product( $product ) ) {
`

and that'l fix the issue.
But of course they have a better knowledge of their own code.
Inside that if block the code handles the missing `$product`.

Just after that if block the code continues with:
`	$tabs    = apply_filters( 'woocommerce_product_tabs', array() );`
which is the WC filter hook that at the end produces the fatal.

**Comment by thomasplevy** [comment]
@nrherron92 if you haven't done so already please alert affected users that we're going to do our best to see if ET can get his resolved from their side and that they could open a support request with ET to let them know they're affected by this issue and that may help get this resolved sooner. It would be good to direct them to this issue so that if ET devs do get alerted they'll have this information to expedite debugging

**Comment by toyinogunseinde** [comment]
Another ticket HS-116450

**Comment by thomasplevy** [comment]
Bumping my contact...

**Comment by andrey-agp** [comment]
another one https://app.intercom.com/a/apps/hrpt54hy/inbox/inbox/2206660/conversations/26515826903

**Comment by toyinogunseinde** [comment]
Another ticket here: HS-117249

**Comment by thomasplevy** [comment]
I have been connected with some people at ET but haven't heard back from them since 4/7. I just sent a little reminder email to check on the status.

**Comment by toyinogunseinde** [comment]
We are seeing an increase in this number HS-117387

**Comment by thomasplevy** [comment]
@toyinogunseinde Please continue letting customers know that we are trying to work with ET to resolve this issue. If you haven't already told them, please ask them to reach out to ET as well to see if they can get any assistance from ET directly instead of relying on us to be solely responsible for resolving this conflict. Thank you.

**Comment by thomasplevy** [comment]
@toyinogunseinde @nrherron92 we're going to have to resolve this from our side, fix pending at #1099, awaiting code review and will hopefully be able to ship tomorrow
